File: management-domain/Sales-department/Backend/Java/communication-service/src/main/java/com/gogidix/sales/communication/shared/requestcontext/RequestContext.java
package com.gogidix.sales.communication.shared.requestcontext;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

/**
 * Request Context
 * Holds tenant and user information for the current request
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RequestContext {

    private String tenantId;
    private String userId;
    private String correlationId;
    private String requestId;
    private String channel;

    private Map<String, Object> metadata;
}

File: management-domain/Customer-support/Customer-support/Backend/Node.js/sentiment-analysis-service/src/shared/context/request-context.ts
export class RequestContext {
  private static _current: RequestContext | null = null;
  constructor(public readonly requestId: string, public readonly userId?: string, public readonly correlationId?: string) {}
  static set(context: RequestContext): void { RequestContext._current = context; }
  static get(): RequestContext | null { return RequestContext._current; }
  static clear(): void { RequestContext._current = null; }
}
